modules/dnn/src/darknet/darknet_io.cpp

```diff
@@ -725,42 +801,64 @@ namespace cv {
                     std::map<std::string, std::string> &layer_params = i->second;
                     std::string layer_type = layer_params["type"];
 
-                    if (layer_type == "convolutional")
+                    if (layer_type == "convolutional" || layer_type == "connected")
                     {
-                        int kernel_size = getParam<int>(layer_params, "size", -1);
-                        int filters = getParam<int>(layer_params, "filters", -1);
-                        bool use_batch_normalize = getParam<int>(layer_params, "batch_normalize", 0) == 1;
+                        size_t weights_size;
+                        int nums;
+                        bool use_batch_normalize;
+                        cv::Mat weightsBlob;
+                        if(layer_type == "convolutional")
+                        {
+                            int kernel_size = getParam<int>(layer_params, "size", -1);
+                            int filters = getParam<int>(layer_params, "filters", -1);
```

Comment:
we can do `nums =  getParam<int>(layer_params, "filters", -1);` directly

Reply:
Let's use `numOutChannels`. We don't need `filters` here

Reply:
On the other side, let's rename `numOutChannels` to `filters`, yes. This way we will have less changes below. Sorry for inconvenience :)

Reply:
> Let's use `numOutChannels`. We don't need `filters` here

ohh Sorry, Couldn't notice that. I will remove this line, 
https://github.com/opencv/opencv/blob/24932c781ea872cf542536402793b43a8d8c0216/modules/dnn/src/darknet/darknet_io.cpp#L821

Reply:
> On the other side, let's rename `numOutChannels` to `filters`, yes. This way we will have less changes below. Sorry for inconvenience :)

you want me to rename `numOutChannels` to `filters` everywhere ? Just wanted to clarify once.

Reply:
No problem. let me change that.

Reply:
Ok, thanks! So let's fix that and we are going to merge PR.

Reply:
Done !